# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

**Lane**: Growth / Recovery / Momentum Prediction (Freestyle).

**Research Question**: Can historical search, engagement, and content signals be used to predict which content pages are likely to experience a significant decline in organic search performance over a future time window?

**Motivation**:

The goal is to investigate whether a machine learning model can rank pages according to future decline risk more effectively than a transparent rule-based baseline. The project focuses on prioritizing pages for human review rather than automatically making SEO decisions. Large websites often contain thousands of pages, making manual review impractical. SEO analysts need to decide which pages deserve attention first because reviewing every page is not feasible.

Observable signals such as search impressions, click-through rate (CTR), average ranking position, content freshness, engagement, and content age may contain useful information about future performance. Rather than relying only on manually designed thresholds, machine learning may identify combinations of signals that improve prioritization.

The objective is not to replace SEO analysts but to help them allocate their limited review time more efficiently.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Decision Supported**: Which pages should an SEO analyst review first because they are most likely to experience future performance decline?

After reviewing a recommended page, an SEO analyst may decide to:

- Refresh outdated content.
- Expand thin or incomplete content.
- Improve page titles or meta descriptions.
- Improve user engagement or content quality.
- Merge, prune, or consolidate overlapping content.
- Continue monitoring without making changes.

The model recommends pages for review only. The final SEO decision remains with the analyst.

**Cost of Incorrect Recommendations**: 

A false positive may cause analysts to spend time reviewing pages that are unlikely to decline. A false negative may cause genuinely at-risk pages to be missed, potentially resulting in continued loss of organic traffic before intervention. Because analyst time is limited, ranking quality is more valuable than maximizing overall classification accuracy.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [1]:
import pandas as pd

df = pd.read_csv("content_refresh_anonymized.csv")
df.head(3)

,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9


In [2]:
trend_counts = df["trend_direction"].value_counts()

print(trend_counts)
print((trend_counts / len(df) * 100).round(1))

trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64
trend_direction
down      54.2
stable    19.9
up        14.6
new        7.5
flat       3.8
Name: count, dtype: float64


The dataset contains `16262` pages (`54.2%`) labeled as `down`, indicating that performance deterioration is common. Given the large number of potentially declining pages, a data-driven ranking system could help SEO analysts prioritize which pages to review first.

In [3]:
df.groupby("trend_direction")["impressions_90d"].mean().round(1)

trend_direction
down      4919.1
flat        20.7
new        164.1
stable    9213.6
up        4716.1
Name: impressions_90d, dtype: float64

In [4]:
df.groupby("trend_direction")["content_age_days"].mean().round(1)

trend_direction
down      236.2
flat      245.9
new       238.7
stable    295.4
up        288.5
Name: content_age_days, dtype: float64

In addition, pages with different trend directions show different average impressions and content ages, suggesting that historical search and content features may contain useful information for prioritizing pages at risk of future decline.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project can identify statistical relationships between historical search, engagement, and content features and future page performance. If the model performs well on unseen data, it may provide useful decision support by ranking pages according to their estimated risk of future decline.

The results should be interpreted as predictive rather than causal. A high predicted risk does not prove that a page will decline or explain why it declines. Likewise, improving a feature such as CTR or updating content does not guarantee improved rankings or traffic.

This project does **not** attempt to predict Google's ranking algorithm, determine the exact causes of ranking changes, or guarantee that a specific SEO action will improve performance. Instead, it aims to help SEO analysts prioritize which pages to review first based on observable historical signals.

Any recommendations produced by the model should be treated as decision-support for human analysts rather than automatic SEO decisions.


## Self-check

Before you submit, confirm each line honestly:

- [X] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.